libnvme/ctrl.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Gris Ge <fge@redhat.com>
+ */
+
+#ifndef _NVME_CTRL_H_
+#define _NVME_CTRL_H_
+
+#include <stdint.h>
+
+#include <libnvme/libnvme.h>
+
+#pragma pack(push, 1)
+struct _nvme_spec_psd {
+	uint8_t mp[2];
+	uint8_t reserved_0;
+	uint8_t _bit_field_0;		/* mxps:1, nops:1, reserve:6 */
+	uint8_t enlat[4];
+	uint8_t exlat[4];
+	uint8_t _bit_field_1;		/* rrt:5, reserve:3 */
+	uint8_t _bit_field_2;		/* rrl:5, reserved:3 */
+	uint8_t _bit_field_3;		/* rwt:5, reserved:3 */
+	uint8_t _bit_field_4;		/* rwl:5, reserved:3 */
+	uint8_t idlp[2];
+	uint8_t _bit_field_5;		/* reserved:6, ips:2 */
+	uint8_t reserved_7;
+	uint8_t actp[2];
+	uint8_t _bit_field_6;		/* apw:3, reserved:3, aps:2 */
+	uint8_t reserved_9[9];
+};
+
+struct _nvme_spec_fabric_id_ctrl_data {
+	uint8_t ioccsz[4];
+	uint8_t iorcsz[4];
+};
+
+struct _nvme_spec_id_ctrl_data {
```

Comment:
As I mentioned in https://github.com/linux-nvme/nvme-cli/pull/533 I'm not fond of having 2 representations of the same data.  Is there anyway we can leverage `struct nvme_id_ctrl` so that we don't have 2 structures to maintain?

Reply:
Will do.